src/librustc_mir/const_eval.rs

```diff
@@ -619,9 +618,7 @@ pub fn const_eval_raw_provider<'a, 'tcx>(
         let tables = tcx.typeck_tables_of(def_id);
 
         // Do match-check before building MIR
-        if let Err(ErrorReported) = tcx.check_match(def_id) {
-            return Err(ErrorHandled::Reported)
-        }
+        tcx.check_match(def_id);
```

Comment:
Calling this feels pointless, also similar for `mir_const_qualif` below.
cc @oli-obk Can we just remove these and rely on `tainted_by_errors` instead?

Reply:
I seem to remember that building MIR before match checking results in ICEs. Not sure this belongs here though.

Reply:
That shouldn't make sense nowadays, though, there should be no mutable state involved?

Reply:
As in `check_match` reports errors for `match` expressions while MIR building ICEs on those same expressions.

Reply:
Ah, sure, but that means MIR building should be handling this gracefully.

Reply:
Are you ok with this being addressed in a separate PR? If so, I'll open an issue

Reply:
It should be handled here, as the behavior of "exit if `check_match` errored" is gone! You could try to turn an error into an ICE if you want.

Reply:
But I guess I'd be fine with a `// FIXME(#...) check_match may have errored but we're not checking for that anymore` here.